repositories/gcs.go

```diff
@@ -49,14 +49,22 @@ func getVersionHistoryFromGCS(onlyFullReleases bool) ([]string, error) {
 	available := getVersionsFromGCSPrefixes(prefixes)
 	sorted := versions.GetInAscendingOrder(available)
 
-	if onlyFullReleases && len(sorted) > 0 {
-		latestVersion := sorted[len(sorted)-1]
-		_, isRelease, err := listDirectoriesInReleaseBucket(latestVersion + "/release/")
-		if err != nil {
-			return []string{}, fmt.Errorf("could not list release candidates for latest release: %v", err)
+	// TODO(#171): This algorithm is incorrect if version == 'latest-n' and any of the last n versions (except the last one) does not have a release yet.
```

Comment:
Just for my understanding, what we would actually want to do here is to basically filter the `sorted` array and remove all entries for which no release version exists, right? But we don't do that, because you think it's too slow due to the number of HTTP requests?

Reply:
Yes, that's the problem. But we can work around this by specifying how many of the latest releases we need: #171